### DISCUSSION TITLE
Are mounted WSGI applications blocking (when using the middleware)?

### DISCUSSION THREAD

**Discussion by septatrix** [encode/starlette#1305] [Q&A] [answered]
We are in the process of migrating our API to FastAPI and would like to mount unmigrated parts using starlettes `WSGIMiddleware`. Based on the code it seems like each request is assigned to a worker in a thread pool however I was unable to find any documentation of the WSGI middleware and would like some clarification.

**Comment by Kludex** [comment]
What would you like to know?

**Reply by septatrix** [reply]
Whether or not multiple request to the wsgi middleware are blocking (i.e. nothing else runs) or if every wsgi connection has its own thread (or similar) which is spawned when the request comes in. If the wsgi middleware is spawning threads this would allow performing blocking actions like subprocesses or fetching data from another server without the whole starlett server stalling.

**Comment by Kludex** [comment, marked as answer]
They are not blocking.
